### PR TITLE
Update changelog and versions for v2.12-beta.1

### PR DESCRIPTION
**Pull request by karimnaaji**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,21 @@
+## 2.12.0-beta.1
+
+## Features ✨ and improvements 🏁
+
+* Improve performance of patterns and line dashes, also improving their appearance when zooming ([#12326](https://github.com/mapbox/mapbox-gl-js/pull/12326))
```
Comment (SnailBones) [inline CHANGELOG.md:5]:
```suggestion
* Improve performance of patterns and line dashes and improve their appearance when zooming. ([#12326](https://github.com/mapbox/mapbox-gl-js/pull/12326))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,21 @@
...
+* Improve performance of text and icon placement. ([#12351](https://github.com/mapbox/mapbox-gl-js/pull/12351))
+* Improve performance of loading terrain data ([#12397](https://github.com/mapbox/mapbox-gl-js/pull/12397))
+* Improve poles support in `getBounds` when projection is set to globe ([#12315](https://github.com/mapbox/mapbox-gl-js/pull/12315))
```
Comment (SnailBones) [inline CHANGELOG.md:8]:
```suggestion
* Improve behavior of `getBounds` at the poles on a globe projection. ([#12315](https://github.com/mapbox/mapbox-gl-js/pull/12315))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,21 @@
...
+* Allow zooming towards terrain at a safe distance without pitching the camera ([#12354](https://github.com/mapbox/mapbox-gl-js/pull/12354))
```
Comment (SnailBones) [inline CHANGELOG.md:9]: Should this be a bugfix? Maybe something like:
```
Fix pitch changing to 0 on zooming in to terrain
```

File: CHANGELOG.md (hunk 2 above)
Comment (SnailBones) [reply CHANGELOG.md]: Also, should we consider this a bugfix?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,21 @@
+## 2.12.0-beta.1
+
+## Features ✨ and improvements 🏁
+
+* Improve performance of patterns and line dashes and improve their appearance when zooming. ([#12326](https://github.com/mapbox/mapbox-gl-js/pull/12326))
+* Improve performance of text and icon placement. ([#12351](https://github.com/mapbox/mapbox-gl-js/pull/12351))
+* Improve performance of loading terrain data ([#12397](https://github.com/mapbox/mapbox-gl-js/pull/12397))
+* Improve behavior of `getBounds` at the poles on a globe projection. ([#12315](https://github.com/mapbox/mapbox-gl-js/pull/12315))
```
Comment (SnailBones) [inline CHANGELOG.md:8]: Should we consider this a bugfix?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,21 @@
...
+* Allow zooming towards terrain at a safe distance without pitching the camera ([#12354](https://github.com/mapbox/mapbox-gl-js/pull/12354))
+* Allow for pitch override in cameraForBounds, `fitBounds` and `fitScreenCoordinates` camera APIs ([#12367](https://github.com/mapbox/mapbox-gl-js/pull/12367))
```
Comment (SnailBones) [inline CHANGELOG.md:10]:
```suggestion
* Allow for pitch override in `cameraForBounds`, `fitBounds` and `fitScreenCoordinates` camera APIs. ([#12367](https://github.com/mapbox/mapbox-gl-js/pull/12367))
```

File: CHANGELOG.md (hunk 3 above)
Comment (karimnaaji) [reply CHANGELOG.md:9]: I was also considering moving it to a bug fix but it seems more fitting under the improvement section as the "right" behavior can be quite subjective.

File: CHANGELOG.md (hunk 4 above)
Comment (karimnaaji) [reply CHANGELOG.md:8]: Yes, let's move that one on bugfix.

File: CHANGELOG.md (hunk 3 above)
Comment (SnailBones) [reply CHANGELOG.md:8]: Good point! My thinking was that it fixed [this bug](https://github.com/mapbox/mapbox-gl-js/issues/10197) but it looks like the scope of that ticket is smaller. Thoughts on rephrasing to:

```suggestion
* Update collisions with terrain to longer adjust pitch and reduce clipping. Mouse panning into terrain now lifts the camera above terrain. ([#12354](https://github.com/mapbox/mapbox-gl-js/pull/12354))
```

cc @avpeery

Comment (avpeery) [reply CHANGELOG.md:8]: I personally considered it a bug but seems like both phrasings and categories fit, so whatever works!

**Pull request merged**